monai/transforms/intensity/array.py

```diff
@@ -1137,3 +1131,158 @@ def __call__(self, img: np.ndarray) -> np.ndarray:
+            If a length-2 list is given as [a,b] then the value of alpha will be
+            sampled uniformly from the interval [a,b]. 0 <= a <= b <= 1.
+        as_tensor_output: if true return torch.Tensor, else return np.array. default: True.
+    """
+
+    def __init__(self, prob: float = 0.1, alpha: Sequence[float] = (0.0, 1.0), as_tensor_output: bool = True) -> None:
+
+        if len(alpha) != 2:
+            raise AssertionError("alpha length must be 2.")
+        if alpha[1] > 1 or alpha[0] < 0:
+            raise AssertionError("alpha must take values in the interval [0,1]")
+        if alpha[0] > alpha[1]:
+            raise AssertionError("When alpha = [a,b] we need a < b.")
+
+        self.alpha = alpha
+        self.sampled_alpha = -1.0  # stores last alpha sampled by randomize()
+        self.as_tensor_output = as_tensor_output
+
+        RandomizableTransform.__init__(self, prob=prob)
+
+    def __call__(self, img: Union[np.ndarray, torch.Tensor]) -> Union[torch.Tensor, np.ndarray]:
+
+        # randomize application and possibly alpha
+        self.randomize(None)
+
+        if self._do_transform:
+            # apply transform
+            transform = GibbsNoise(self.sampled_alpha, self.as_tensor_output)
+            img = transform(img)
+        else:
+            if isinstance(img, np.ndarray) and self.as_tensor_output:
+                img = torch.Tensor(img)
+            elif isinstance(img, torch.Tensor) and not self.as_tensor_output:
+                img = img.detach().cpu().numpy()
+        return img
+
+    def randomize(self, _: Any) -> None:
+        """
+        (1) Set random variable to apply the transform.
+        (2) Get alpha from uniform distribution.
+        """
+        super().randomize(None)
+        self.sampled_alpha = self.R.uniform(self.alpha[0], self.alpha[1])
+
+
+class GibbsNoise(Transform):
+    """
+    The transform applies Gibbs noise to 2D/3D MRI images. Gibbs artifacts
+    are one of the common type of type artifacts appearing in MRI scans.
+
+    The transform is applied to all the channels in the data.
+
+    For general information on Gibbs artifacts, please refer to:
+    https://pubs.rsna.org/doi/full/10.1148/rg.313105115
+    https://pubs.rsna.org/doi/full/10.1148/radiographics.22.4.g02jl14949
+
+
+    Args:
+        alpha (float): Parametrizes the intensity of the Gibbs noise filter applied. Takes
+            values in the interval [0,1] with alpha = 0 acting as the identity mapping.
+        as_tensor_output: if true return torch.Tensor, else return np.array. default: True.
+
+    """
+
+    def __init__(self, alpha: float = 0.5, as_tensor_output: bool = True) -> None:
+
+        if alpha > 1 or alpha < 0:
+            raise AssertionError("alpha must take values in the interval [0,1].")
+        self.alpha = alpha
+        self.as_tensor_output = as_tensor_output
+        self._device = "cpu"
```

Comment:
Did you add this or did I? Doesn't seem we're using it.

Reply:
I did.  The operations in `GibbsNoise` are now done with numpy because we saw failures due to Pytorch 1.6 lacking some of the fourier functions. So I save the device of the input tensor in `_device` before going to numpy. Then I make sure to put the tensor to the its original device at the end of `__call___` via `return torch.Tensor(img).to(self._device) if self.as_tensor_output else img`.

Reply:
Ah, I see. Great.